sqlmesh/core/renderer.py

```diff
@@ -206,19 +209,20 @@ def _resolve_tables(
 
         expression = expression.copy()
         with _normalize_and_quote(expression, self._dialect) as expression:
-            snapshots = snapshots or {}
+            snapshots = snapshots or []
             table_mapping = table_mapping or {}
-            mapping = {**to_table_mapping(snapshots.values(), deployability_index), **table_mapping}
-            expand = {d.normalize_model_name(name, dialect=self._dialect) for name in expand} | {
-                name for name, snapshot in snapshots.items() if snapshot.is_embedded
-            }
+            mapping = {**to_table_mapping(snapshots, deployability_index), **table_mapping}
+            expand = set(expand) | {snapshot.fqn for snapshot in snapshots if snapshot.is_embedded}
 
             if expand:
+                model_mapping = {
+                    snapshot.fqn: snapshot.model for snapshot in snapshots if snapshot.is_model
+                }
 
                 def _expand(node: exp.Expression) -> exp.Expression:
                     if isinstance(node, exp.Table) and snapshots:
                         name = exp.table_name(node)
-                        model = snapshots[name].model if name in snapshots else None
+                        model = model_mapping.get(name)
```

Comment:
How do we know that this name is FQN?

Reply:
I guess `_normalize_and_quote` qualifies tables. Please confirm

Reply:
Yes all table references in the query should be fully qualified at this point therefore it is an fqn.